ingest/ingest.py

```diff
@@ -18,162 +18,247 @@
+        contributor['userType'] = contributorJson['type']
         if(contributorJson['type']=='User'):
             contributor['username'] = contributorJson['login']
-            contributor['profile_url'] = contributorJson['html_url']
-            contributor['avatar_url'] = contributorJson['avatar_url']
+            contributor['profileUrl'] = contributorJson['html_url']
+            contributor['avatarUrl'] = contributorJson['avatar_url']
         else:
             contributor['username'] = ''
-            contributor['profile_url'] = ''
-            contributor['avatar_url'] = ''
+            contributor['profileUrl'] = ''
+            contributor['avatarUrl'] = ''
         contributors.append(contributor)
     contributionMap = {'commitTotal': commitTotal, 'contributorsMap': contributors}
     return contributionMap
 
-def get_github_property(repo, property_name):
-    org = repo['owner']['login']
-    name = repo['name']
-    return requests.get('https://api.github.com/repos/' + org + '/' + name + '/' + property_name + '?access_token='+os.environ['GITHUB_ACCESS_TOKEN']).text
-
-def clone_repos(repos):
-  for repo in repos:
-    setup_cloning_dir(repo['org'], repo['project_name'])
-    call(["git","clone",repo['clone_url']])
-
-def setup_cloning_dir(org, repo):
-    orgdir = cloned_project_path + '/' + org
-    repodir = orgdir + '/' + repo
+def getForks(forksJson):
+    result = []
+    for fork in forksJson:
+        repo = {}
+        repo['id'] = str(fork['owner']['id']) + '-' + str(fork['id'])
+        repo['name'] = str(fork['name'])
+        repo['owner'] = str(fork['owner']['login'])
+        result.append(repo)
+    return result
+
+def getGithubReadmeObject(readmeObj):
+    result = {}
+    if 'content' in readmeObj:
+        decoded_readme = str(base64.b64decode(readmeObj['content']),'utf-8')
+        result['content'] = decoded_readme
+        result['url'] = readmeObj['url']
+    else:
+        repo['readmeRaw'] = {'content': '', 'url': ''}
+
+    return result
+
+def getReleases(releasesJson):
+    results = []
+    for release in releasesJson:
+        result = {}
+        result['tagName'] = release['tag_name']
+        result['name'] = release['name']
+        result['id'] = release['id']
+        # none of the test cases currently have assets or download counts. Will test this later
+        result['assets'] = []
+        result['downloads'] = 0
+        results.append(result)
+    return results
+
+def calculateRank(repo):
+    result = (repo['sourceData']['stars']*3) + (repo['sourceData']['watchers']*4) + (len(repo['sourceData']['contributors'])*5) + repo['sourceData']['commits']
+    return result
+
+def cloneGithubRepo(repo):
+    ownerName = repo['sourceData']['owner']['name']
+    repoName = repo['sourceData']['name']
+
+    rootdir = expanduser("~") + '/cloned_projects'
+    repodir = rootdir + '/' + ownerName + '/' + repoName
+    # if os.path.exists(repodir):
```

Comment:
Lets remove unused code or add a comment to why it is commented out

Reply:
Done